### PR TITLE
Reducer updates.

### PR DESCRIPTION
**Pull request by oraluben**
This PR:
1. Add information when reproducer (and reducer) is not available. This will help the following reducer implementations;
2. Illustrate how to add reducer to a oracle class (`TiDBTLPWhereOracle`);
3. Add doc about reproducer.

Some minor changes:
1. Bump `actions/setup-java` to v3 to support caching maven artifacts;
2. Remove non-LTS JDK tests. This currently blockes merging because those tests are marked as required, which is no longer necessary IMO. I suggest to remove them from required tests.

### CODE REVIEW

File: src/sqlancer/Main.java
Comment (mrigger) [inline src/sqlancer/Main.java:455]: I think this should be "does not support" since the message is about a specific oracle. Other than that, the PR looks good to me.

**Review by mrigger — APPROVED** [review]
LGTM! @bajinsheng, could you perhaps have a quick look at the TiDB reducer?

File: src/sqlancer/Main.java
Comment (bajinsheng) [inline src/sqlancer/Main.java:454]: Why write this error information into the query plan logging file?

File: src/sqlancer/tidb/oracle/TiDBTLPWhereOracle.java
Comment (bajinsheng) [inline src/sqlancer/tidb/oracle/TiDBTLPWhereOracle.java:48]: Can you give more documents about how to support reproducers for a new DBMS? For this code, I am not very sure how it works, and it seems different from the reproducer in the NoREC oracle.

Comment (bajinsheng) [reply src/sqlancer/tidb/oracle/TiDBTLPWhereOracle.java:48]: Can you change it to a unified format as the reproducer in NoREC?

**Comment by mrigger** [comment]
Let me know when/if I can merge the PR. On vacation next week.

**Comment by oraluben** [comment]
I think it's good to go, let me know if I've answered all your concerns @bajinsheng

**Comment by bajinsheng** [comment]
@oraluben Please see my comment above. I am just wondering can you make the coding style consistent?

**Comment by mrigger** [comment]
Hmm, should we merge the code to have the extra functionality? Perhaps we can refactor the other code to use a lambda as well as part of a subsequent PR.

**Comment by oraluben** [comment]
> I am just wondering can you make the coding style consistent?

Do you mean use lambda v.s. define new class for each reducer? I think there's no fundamental difference between them as reducer implementation, and can be chosen accordingly to type of reproducer.

**Comment by bajinsheng** [comment]
@oraluben These are definitely semantically equivalent implements, but it is not a good coding style.

@mrigger  I am fine to merge it since the PR has been open for a quite long time.

**Comment by oraluben** [comment]
I think we can merge it first, and unify the style later.

**Comment by mrigger** [comment]
Currently, the style checks are failing. Could you please execute `mvn formatter:format`?

**Comment by mrigger** [comment]
Checkstyle is failing (see `mxn verify -DskipTests=true`). After fixing this, we can merge the PR. Thanks!

**Comment by mrigger** [comment]
Sorry, one last thing: I noticed that you accidentally checked in some binary files (in the `databases` subdirectory). Could you remove those?

**Comment by oraluben** [comment]
> Sorry, one last thing: I noticed that you accidentally checked in some binary files (in the `databases` subdirectory). Could you remove those?

Oops, sorry for that, now it should be ok

**Pull request merged**
